### PR TITLE
expression, executor: rewrite builtin func Sin

### PR DESCRIPTION
**Pull request by plutolove**

### CODE REVIEW

File: expression/builtin_math_test.go [hunk 1]
```diff
@@ -532,30 +533,45 @@ func (s *testEvaluatorSuite) TestRadians(c *C) {
...
-		{nil, nil},
-		{int64(0), float64(0)},
-		{math.Pi, float64(math.Sin(math.Pi))}, // Pie ==> 0
-		{-math.Pi, float64(math.Sin(-math.Pi))},
-		{math.Pi / 2, float64(math.Sin(math.Pi / 2))}, // Pie/2 ==> 1
-		{-math.Pi / 2, float64(math.Sin(-math.Pi / 2))},
-		{math.Pi / 6, float64(math.Sin(math.Pi / 6))}, // Pie/6(30 degrees) ==> 0.5
-		{-math.Pi / 6, float64(math.Sin(-math.Pi / 6))},
-		{math.Pi * 2, float64(math.Sin(math.Pi * 2))},
-		{"0.000", float64(0)}, // string value case
```
Comment (XuHuaiyu) [inline expression/builtin_math_test.go]: why remove this?

File: expression/builtin_math.go [hunk 2]
```diff
@@ -1044,32 +1045,30 @@ type sinFunctionClass struct {
 }
 
 func (c *sinFunctionClass) getFunction(args []Expression, ctx context.Context) (builtinFunc, error) {
-	sig := &builtinSinSig{newBaseBuiltinFunc(args, ctx)}
+	tp := types.NewFieldType(mysql.TypeDouble)
+	tp.Flen = 8
```
Comment (XuHuaiyu) [inline expression/builtin_math.go]: same as the `cos` PR

File: expression/builtin_math_test.go (hunk 1 above)
Comment (plutolove) [inline expression/builtin_math_test.go]: just rewrite

File: expression/builtin_math_test.go [hunk 3]
```diff
@@ -532,30 +533,46 @@ func (s *testEvaluatorSuite) TestRadians(c *C) {
...
+		{nil, 0, true, false},
+		{int64(0), float64(0), false, false},
+		{math.Pi, float64(math.Sin(math.Pi)), false, false}, // Pie ==> 0
+		{-math.Pi, float64(math.Sin(-math.Pi)), false, false},
+		{math.Pi / 2, float64(math.Sin(math.Pi / 2)), false, false}, // Pie/2 ==> 1
+		{-math.Pi / 2, float64(math.Sin(-math.Pi / 2)), false, false},
+		{math.Pi / 6, float64(math.Sin(math.Pi / 6)), false, false}, // Pie/6(30 degrees) ==> 0.5
+		{-math.Pi / 6, float64(math.Sin(-math.Pi / 6)), false, false},
+		{math.Pi * 2, float64(math.Sin(math.Pi * 2)), false, false},
+		{string("adfsdfgs"), 0, false, true},
```
Comment (XuHuaiyu) [inline expression/builtin_math_test.go]: what error will we get here?

Comment (plutolove) [inline expression/builtin_math_test.go]: can not parse the "adfsdfgs" to a float64 number

Comment (XuHuaiyu) [inline expression/builtin_math_test.go]: ok, I got it.

**Comment by XuHuaiyu** [comment]
LGTM

**Comment by winkyao** [comment]
@plutolove please fix conflicts

File: expression/builtin_math.go [hunk 4]
```diff
@@ -1021,7 +1021,7 @@ type builtinRadiansSig struct {
 	baseBuiltinFunc
 }
 
-// eval evals a builtinRadiansSig.
+// evalreal evals a builtinRadiansSig.
```
Comment (winoros) [inline expression/builtin_math.go]: This change should be applied to line 1066?

**Pull request merged**